### PR TITLE
🎓 Strings Topic For Loops

### PR DESCRIPTION
**Pull request by jameshughes89**
### What
Show for loops,

### Additional Notes
No tests are included for the vertical prints. Any workaround to test the standard out will be confusing AF.

### CODE REVIEW

File: site/topics/strings/strings.rst [hunk 1]
```diff
@@ -35,60 +35,90 @@ Strings & Objects
...
+
+
+* Both the``while`` and ``for`` loops are perfectly fine for this situation
+
+    * But you may find the ``for`` loop has a little nicer syntax
+
+.. note::
+
+    In the ``vertical_print_for`` example, the use of the variable name ``char`` in the ``for`` loop is arbitrary. It is
+    justa variable name and is not required to be ``char`` simply because the things in the string are characters. I
```
Comment (twentylemon) [inline site/topics/strings/strings.rst]: justa -> just a

File: site/topics/strings/strings.rst [hunk 2]
```diff
@@ -35,60 +35,90 @@ Strings & Objects
...
+    just chose ``char`` since it is an appropriate name for the variable. For example, the following code would work
+    just fine:
```
Comment (twentylemon) [inline site/topics/strings/strings.rst:113]: overusing _just_ imo

**Review by twentylemon — COMMENTED** [review]
> No tests are included for the vertical prints. Any workaround to test the standard out will be confusing AF.

Capturing output is fairly straightforward in unittest (it's much simpler in pytest, but here we are). Mock out `builtins.println` and verify behaviour. For example:

```python
@unittest.mock.patch("builtins.println")
def check_it_brother(println_mock):
    for c in "Hello World":
        println(c)
    println_mock.assert_has_calls([
        unittest.mock.call('H'), 
        # ...etc
    ])
```

You can alternitively patch out `sys.stdout`
```python
@unittest.mock.patch("sys.stdout", new_callable=io.StringIO)
def capture_me(stdout_mock):
    func()
    self.assertEqual("expected output", stdout_mock.getvalue())
```

File: site/topics/strings/strings.rst [hunk 3]
```diff
@@ -108,8 +108,8 @@ For loops
 .. note::
 
     In the ``vertical_print_for`` example, the use of the variable name ``char`` in the ``for`` loop is arbitrary. It is
-    just a variable name and is not required to be ``char`` simply because the things in the string are characters. I
-    just chose ``char`` since it is an appropriate name for the variable. For example, the following code would work
+    just a variable name and is not required to be ``char`` because the things in the string are characters. I just
+    chose ``char`` since it is an appropriate name for the variable. For example, the following code would work
```
Comment (jameshughes89) [inline site/topics/strings/strings.rst:112]: I forgot to push this one.

**Comment by jameshughes89** [comment]
> > No tests are included for the vertical prints. Any workaround to test the standard out will be confusing AF.
> 
> Capturing output is fairly straightforward in unittest (it's much simpler in pytest, but here we are). Mock out `builtins.println` and verify behaviour. For example:
> 
> ```python
> @unittest.mock.patch("builtins.println")
> def check_it_brother(println_mock):
>     for c in "Hello World":
>         println(c)
>     println_mock.assert_has_calls([
>         unittest.mock.call('H'), 
>         # ...etc
>     ])
> ```
> 
> You can alternitively patch out `sys.stdout`
> 
> ```python
> @unittest.mock.patch("sys.stdout", new_callable=io.StringIO)
> def capture_me(stdout_mock):
>     func()
>     self.assertEqual("expected output", stdout_mock.getvalue())
> ```

noice

File: test/test_strings_topic.py [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+class TestStringsTopic(unittest.TestCase):
+    @unittest.mock.patch("sys.stdout", new_callable=io.StringIO)
+    def test_vertical_print_while_empty_string_prints_nothing(self, mock_stdout):
+        vertical_print_while("")
+        expected = ""
+        self.assertEqual(expected, mock_stdout.getvalue())
+
+    @unittest.mock.patch("sys.stdout", new_callable=io.StringIO)
+    def test_vertical_print_while_hello_string_prints_nothing(self, mock_stdout):
```
Comment (twentylemon) [inline test/test_strings_topic.py:16]: test names are wrong -- prints_nothing

**Review by twentylemon — APPROVED** [review]

**Pull request merged**